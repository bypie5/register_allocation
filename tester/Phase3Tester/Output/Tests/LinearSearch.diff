--- SelfTestCases/LinearSearch.out	2020-03-05 22:31:29.000000000 -0800
+++ ./Output/Tests/LinearSearch.out	2020-03-09 14:31:07.000000000 -0700
@@ -1,15 +1,5 @@
-10
-11
-12
-13
-14
-15
-16
-17
-18
-9999
-0
-1
-1
-0
-55
+--- Error (ProgramFault) ---
+require function address, found integer10
+  Main() line 20
+  LS.Start() line 39
+  LS.Init() line 189
