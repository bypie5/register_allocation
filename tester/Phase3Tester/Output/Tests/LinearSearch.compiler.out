const vmt_LS
    :LS.Start
    :LS.Print
    :LS.Search
    :LS.Init

func Main [in 0, out 0, local 3]
local[0] = $s0
local[1] = $s1
local[2] = $s2
$s0 = HeapAllocZ(12)
[$s0+0] = :vmt_LS
if $s0 goto :null1
Error("null pointer")
null1:
$s1 = [$s0+0]
$s1 = [$s1+0]
$a0 = $s0
$a1 = 10
call $s1
$s2 = $v0
PrintIntS($s2)
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
ret

func LS.Start [in 0, out 0, local 4]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
$s1 = $a0
$s0 = $a1
$s2 = [$s1+0]
$s2 = [$s2+12]
$a0 = $s1
$a1 = $s0
call $s2
$s3 = $v0
$s0 = [$s1+0]
$s0 = [$s0+4]
$a0 = $s1
call $s0
$s2 = $v0
PrintIntS(9999)
$s0 = [$s1+0]
$s0 = [$s0+8]
$a0 = $s1
$a1 = 8
call $s0
$s2 = $v0
PrintIntS($s2)
$s0 = [$s1+0]
$s0 = [$s0+8]
$a0 = $s1
$a1 = 12
call $s0
$s2 = $v0
PrintIntS($s2)
$s0 = [$s1+0]
$s0 = [$s0+8]
$a0 = $s1
$a1 = 17
call $s0
$s2 = $v0
PrintIntS($s2)
$s0 = [$s1+0]
$s0 = [$s0+8]
$a0 = $s1
$a1 = 50
call $s0
$s2 = $v0
PrintIntS($s2)
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
ret

func LS.Print [in 0, out 0, local 4]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
$s0 = $a0
$s1 = 1
while1_top:
$s2 = [$s0+8]
$s3 = LtS($s1 $s2)
if0 $s3 goto :while1_end
$s2 = [$s0+4]
if $s2 goto :null2
Error("null pointer")
null2:
$s3 = [$s2+0]
$s3 = Lt($s1 $s3)
if $s3 goto :bounds1
Error("array index out of bounds")
bounds1:
$s3 = MulS($s1 4)
$s3 = Add($s3 $s2)
$s2 = [$s3+4]
PrintIntS($s2)
$s1 = Add($s1 1)
goto :while1_top
while1_end:
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
ret

func LS.Search [in 0, out 0, local 7]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
$s1 = $a0
$s0 = $a1
$s2 = 1
$s3 = 0
$s3 = 0
while2_top:
$s4 = [$s1+8]
$s5 = LtS($s2 $s4)
if0 $s5 goto :while2_end
$s4 = [$s1+4]
if $s4 goto :null3
Error("null pointer")
null3:
$s5 = [$s4+0]
$s5 = Lt($s2 $s5)
if $s5 goto :bounds2
Error("array index out of bounds")
bounds2:
$s5 = MulS($s2 4)
$s5 = Add($s5 $s4)
$s4 = [$s5+4]
$s5 = Add($s0 1)
$s6 = LtS($s4 $s0)
if0 $s6 goto :if1_else
$s6 = 0
goto :if1_end
if1_else:
$s6 = LtS($s4 $s5)
$s4 = Sub(1 $s6)
if0 $s4 goto :if2_else
$s4 = 0
goto :if2_end
if2_else:
$s2 = 1
$s2 = 1
$s3 = [$s1+8]
if2_end:
if1_end:
$s3 = Add($s3 1)
goto :while2_top
while2_end:
$v0 = $s2
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
ret

func LS.Init [in 0, out 0, local 7]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
$s2 = $a0
$s1 = $a1
[$s2+8] = $s1
$a0 = $s1
call $s0
$s3 = $v0
[$s2+4] = $s3
$s0 = 1
$s1 = [$s2+8]
$s3 = Add($s1 1)
while3_top:
$s1 = [$s2+8]
$s4 = LtS($s0 $s1)
if0 $s4 goto :while3_end
$s1 = MulS(2 $s0)
$s4 = Sub($s3 3)
$s5 = [$s2+4]
if $s5 goto :null4
Error("null pointer")
null4:
$s6 = [$s5+0]
$s6 = Lt($s0 $s6)
if $s6 goto :bounds3
Error("array index out of bounds")
bounds3:
$s6 = MulS($s0 4)
$s6 = Add($s6 $s5)
$s5 = Add($s1 $s4)
[$s6+4] = $s5
$s0 = Add($s0 1)
$s3 = Sub($s3 1)
goto :while3_top
while3_end:
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
ret

func AllocArray [in 0, out 0, local 3]
local[0] = $s0
local[1] = $s1
local[2] = $s2
$s0 = $a0
$s1 = MulS($s0 4)
$s1 = Add($s1 4)
$s2 = HeapAllocZ($s1)
[$s2+0] = $s0
$v0 = $s2
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
ret

